src/hotspot/share/jfr/recorder/service/jfrRecorderService.cpp

```diff
@@ -264,6 +264,42 @@ static u4 invoke_with_flush_event(Functor& f) {
   return elements;
 }
 
+class ObjectSamplerStackTraceRepository : public StackObj {
```

Comment:
This class allows `JfrRecorder` to write stack traces from the leak profiler to the current chunk.